include/pagmo/problems/cec2013.hpp

```diff
@@ -0,0 +1,1218 @@
+                retval.append("(schwefel_func_non_rotated)");
+                break;
+            case 15:
+                retval.append("(schwefel_func)");
+                break;
+            case 16:
+                retval.append("(katsuura_func)");
+                break;
+            case 17:
+                retval.append("(bi_rastrigin_func_non_rotated)");
+                break;
+            case 18:
+                retval.append("(bi_rastrigin_func)");
+                break;
+            case 19:
+                retval.append("(grie_rosen_func)");
+                break;
+            case 20:
+                retval.append("(escaffer6_func)");
+                break;
+            case 21:
+                retval.append("(cf01)");
+                break;
+            case 22:
+                retval.append("(cf02)");
+                break;
+            case 23:
+                retval.append("(cf03)");
+                break;
+            case 24:
+                retval.append("(cf04)");
+                break;
+            case 25:
+                retval.append("(cf05)");
+                break;
+            case 26:
+                retval.append("(cf06)");
+                break;
+            case 27:
+                retval.append("(cf07)");
+                break;
+            case 28:
+                retval.append("(cf08)");
+                break;
+        }
+        return retval;
+    }
+    /// Object serialization
+    /**
+     * This method will save/load \p this into the archive \p ar.
+     *
+     * @param ar target archive.
+     *
+     * @throws unspecified any exception thrown by the serialization of the UDP and of primitive types.
+     */
+    template <typename Archive>
+    void serialize(Archive &ar)
+    {
+        ar(m_prob_id, m_rotation_matrix, m_origin_shift, m_y, m_z);
+    }
+
+private:
+    void sphere_func(const double *x, double *f, const vector_double::size_type nx, const double *Os, const double *Mr,
+                     int r_flag) const /* Sphere */
+    {
+        shiftfunc(x, &m_y[0], nx, Os);
+        if (r_flag == 1)
+            rotatefunc(&m_y[0], &m_z[0], nx, Mr);
+        else
+            for (unsigned int i = 0; i < nx; ++i)
+                m_z[i] = m_y[i];
+        f[0] = 0.0;
+        for (unsigned int i = 0; i < nx; ++i) {
+            f[0] += m_z[i] * m_z[i];
+        }
+    }
+
+    void ellips_func(const double *x, double *f, const vector_double::size_type nx, const double *Os, const double *Mr,
+                     int r_flag) const /* Ellipsoidal */
+    {
+        unsigned int i;
+        shiftfunc(x, &m_y[0], nx, Os);
+        if (r_flag == 1)
+            rotatefunc(&m_y[0], &m_z[0], nx, Mr);
+        else
+            for (i = 0; i < nx; ++i)
+                m_z[i] = m_y[i];
+        oszfunc(&m_z[0], &m_y[0], nx);
+        f[0] = 0.0;
+        for (i = 0; i < nx; ++i) {
+            f[0] += std::pow(10.0, 6.0 * i / (nx - 1u)) * m_y[i] * m_y[i];
```

Comment:
I am assuming the behaviour of truncated integer division in ``i / (nx - 1u)`` is intended?